sweagent/environment/swe_env.py

```diff
@@ -545,29 +618,31 @@ def close(self) -> None:
                 self.logger.warning("Failed to remove container", exc_info=True)
             else:
                 self.logger.info("Agent container stopped")
+        self.container_obj = None
         for hook in self.hooks:
             hook.on_close()
 
     # MARK: Helper functions #
 
-    def _reset_container(self) -> None:
-        if self.container is not None:
-            try:
-                self.container.terminate()
-            except KeyboardInterrupt:
-                raise
-            except:
-                self.logger.warning("Failed to terminate container", exc_info=True)
-            else:
-                self.logger.debug("Terminated container")
-        self._init_container()
-        self._init_scripts()
+    # def _reset_container(self) -> None:
```

Comment:
Such a horrid horrid function. Causing a lot of pain: We start a new container when trying to shut down. And when this is not called, we'd have an incompletely initialized container 🤷